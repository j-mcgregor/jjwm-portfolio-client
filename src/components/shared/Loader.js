import React from 'react';
import PropTypes from 'prop-types';
import Pikachu from '../../images/pikachu.gif';
import Pokeball from '../../images/pokeball shake.gif';

const Loader = ({ width, name }) => {
  let type;
  switch (name) {
    case 'pikachu':
      type = Pikachu;
      break;
    case 'pokeball':
      type = Pokeball;
      break;
    default:
      type = '';
      break;
  }

  return (
    <div>
      <img src={type} alt="Loading..." width={width} />
    </div>
  );
};

Loader.propTypes = {
  width: PropTypes.string,
  name: PropTypes.string
};

Loader.defaultProps = {
  width: '',
  name: ''
};

export default Loader;
