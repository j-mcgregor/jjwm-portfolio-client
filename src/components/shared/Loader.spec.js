import React from 'react';
import { shallow } from 'enzyme';
import Loader from './Loader';
import Pikachu from '../../images/pikachu.gif';
import Pokeball from '../../images/pokeball shake.gif';

describe('Loader', () => {
  it('should render the Pikachu gif if name = pikachu', () => {
    // Assemble
    const props = {
      width: '',
      name: 'pikachu'
    };
    const wrapper = shallow(<Loader {...props} />);
    // Assert
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('img').prop('src')).toEqual(Pikachu);
  });
  it('should render the Pokeball gif if name = pokeball', () => {
    // Assemble
    const props = {
      width: '',
      name: 'pokeball'
    };
    const wrapper = shallow(<Loader {...props} />);
    // Assert
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('img').prop('src')).toEqual(Pokeball);
  });
});
