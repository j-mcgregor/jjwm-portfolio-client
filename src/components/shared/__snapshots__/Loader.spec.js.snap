// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render the Pikachu gif if name = pikachu 1`] = `ShallowWrapper {}`;

exports[`Loader should render the Pokeball gif if name = pokeball 1`] = `ShallowWrapper {}`;
